packetbeat/protos/dns/dns.go

```diff
@@ -707,3 +714,160 @@ func decodeDnsPacket(data []byte) (dns *layers.DNS, err error) {
 	}
 	return d, nil
 }
+
+// TCP implementation
+
+func (dns *Dns) Parse(pkt *protos.Packet, tcpTuple *common.TcpTuple, dir uint8, private protos.ProtocolData) protos.ProtocolData {
+	defer logp.Recover("DNS ParseTcp")
+
+	logp.Debug("dns", "Parsing packet addressed with %s of length %d.",
+		pkt.Tuple.String(), len(pkt.Payload))
+
+	priv := dnsPrivateData{}
+
+	if private != nil {
+		var ok bool
+		priv, ok = private.(dnsPrivateData)
+		if !ok {
+			priv = dnsPrivateData{}
+		}
+	}
+
+	payload := make([]byte, 0)
```

Comment:
The [preferred way](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices) to declare and empty slice is `var payload []byte` because it avoids an allocation if the slice is never appended.